src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -175,6 +176,44 @@ private void AddExceptions(ISymbol symbol)
                 }
             }
 
+            protected abstract void AddCaptures(ISymbol symbol);
+
+            protected void AddCaptures(SyntaxNode syntax)
+            {
+                var semanticModel = GetSemanticModel(syntax.SyntaxTree);
+                if (semanticModel.IsSpeculativeSemanticModel)
+                {
+                    // Speculative semantic models cannot do region analysis
+                    return;
+                }
+
+                var analysis = semanticModel.AnalyzeDataFlow(syntax);
+                var captures = analysis.CapturedInside;
+                if (!captures.IsEmpty)
+                {
+                    var parts = new List<SymbolDisplayPart>();
+                    parts.Add(new SymbolDisplayPart(kind: SymbolDisplayPartKind.Text, symbol: null, text: $"\r\n{WorkspacesResources.Captures_colon}"));
+                    bool first = true;
+                    foreach (var captured in captures)
+                    {
+                        if (!first)
+                        {
+                            parts.AddRange(Punctuation(","));
+                        }
+
+                        parts.AddRange(Space(count: 1));
+                        parts.AddRange(ToMinimalDisplayParts(captured, s_formatForCaptures));
```

Comment:
I thinkt the better thing would be to use a format that asks to not escape keywords.  Alternatively, if 'this' is captured, just eplicitly write in the keyword yourself.

Reply:
For the present situation, I'd rather fix the common machinery. "@this" and "this" should both be supported, depending on the symbol.

---
In reply to: [160014322](https://github.com/dotnet/roslyn/pull/23970#discussion_r160014322) [](ancestors = 160014322)

Reply:
Works for me.